backend/otel/otel.go

```diff
@@ -409,8 +415,13 @@ func (o *Handler) HandleLog(w http.ResponseWriter, r *http.Request) {
 					continue
 				}
 
+				timestamp := fields.timestamp
+				if timestamp.Before(minTime) || timestamp.After(maxTime) {
+					timestamp = curTime
+				}
+
 				logRow := clickhouse.NewLogRow(
-					fields.timestamp, uint32(fields.projectIDInt),
+					timestamp, uint32(fields.projectIDInt),
```

Comment:
a lot of our logs also come from the trace log events (see `HandleTrace` above). could we make this timestamp logic filter there too, perhaps by moving it into some common method? maybe `extractFields` could update the timestamp?